js/content/community.js

```diff
@@ -183,6 +182,115 @@ let SpamCommentHandler = (function(){
         observer.observe(modalWait, {attributes: true});
     };
 
+    function updateFavs(favs, emoticonPopup, favBox, favRemove) {
+        LocalStorage.set("fav_emoticons", favs);
+
+        let favsHtml = buildFavBox(favs);
+        HTML.inner(favBox, favsHtml);
+        favBox.querySelectorAll(".emoticon_option").forEach(node => {
+            node.draggable = true;
+            node.querySelector("img").draggable = false;
+            node.addEventListener("dragstart", (ev) => dragFavEmoticon(ev));
+            node.addEventListener("click", (ev) => clickFavEmoticon(ev, emoticonPopup, favRemove))
+        });
+    }
+
+    function dragFavEmoticon(ev) {
+        ev.dataTransfer.setData("emoticon", ev.target.dataset.emoticon);
+    }
+
+    function clickFavEmoticon(ev, emoticonPopup, favRemove) {
+        let name = ev.target.closest(".emoticon_option").dataset.emoticon;
+        let noFav = emoticonPopup.querySelector(`[data-emoticon=${name}]:not(.es_fav)`);
+        noFav.click();
+    }
+
+    function buildFavBox(favs=[]) {
+        let favsHtml = favs.map(fav => buildEmoticonOption(fav)).join("")
+        if (favs.length === 0) {
+            favsHtml = Localization.str.fav_emoticons_dragging;
+        }
+        return favsHtml;
+    }
+
+    function buildEmoticonOption(name) {
+        return `<div class="emoticon_option es_fav" data-emoticon="${name}"><img src="https://steamcommunity-a.akamaihd.net/economy/emoticon/${name}" class="emoticon"></div>`;
```

Comment:
Is it possible to reuse existing emoticon options, since only one of the favorite emoticons can be changed at a time?